### PR TITLE
#167164940 create user signup form

### PR DESCRIPTION
**Pull request by WilliamsOhworuka**
#### Pivotal tracker story
[#167164940 ]https://www.pivotaltracker.com/story/show/167164940 )

#### What does this PR do?
- Created modal component
- Created an input component
- created signup form component

#### Summary of Task
- User should be able signup

#### How can this be tested?
- Git clone
- Run ``` npm install ```
- Run ``` npm run dev ```

#### Screenshots (if appropriate)
Screenshot 2019-08-30 at 8 21 30 AM
Screenshot 2019-08-30 at 8 21 59 AM

### CODE REVIEW

File: src/components/App.js [hunk 1]
```diff
@@ -1,19 +1,21 @@
...
     <h1>Authors Haven</h1>
-    <div className={style.sassyDiv}>Welcome!</div>
+    <div className="sassyDiv">Welcome!</div>
   </div>
 );
 
 const App = () => (
   <Switch>
     <Route exact path="/" component={Welcome} />
     <Route path="/hello" component={HelloDahlia} />
```
Comment (Nkemjiks) [inline src/components/App.js]: Take out this line and dummy component

File: src/components/App.js [hunk 2]
```diff
@@ -1,19 +1,21 @@
...
+    <Route path="/backdrop" component={SignupForm} />
```
Comment (Nkemjiks) [inline src/components/App.js]: This component doesn't need a route

File: src/components/AuthModal/AuthModal.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+/* eslint-disable jsx-a11y/interactive-supports-focus */
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import './AuthModal.scss';
+
+const AuthModal = ({
+  children, title, desc, isOpen,
+}) => {
+  const [display, setDisplay] = useState(isOpen);
```
Comment (Nkemjiks) [inline src/components/AuthModal/AuthModal.js]: Move to a separate js file

File: src/components/AuthModal/AuthModal.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+  const closeModal = () => {
```
Comment (Nkemjiks) [inline src/components/AuthModal/AuthModal.js]: Same here

File: src/components/AuthModal/AuthModal.scss [hunk 5]
```diff
@@ -0,0 +1,80 @@
+.backdrop-div {
```
Comment (Nkemjiks) [inline src/components/AuthModal/AuthModal.scss]: Wrap it all in a parent css

File: src/components/SignupForm/SignupForm.js [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+          setResourceLoading(false);
+        });
+    }
+    event.preventDefault();
+  };
+
+  return (
+    <Aux>
+      {!reqSuccess ? (
+        <AuthModal desc="Sign up to create an account" title="CRAETE AN ACCOUNT" isOpen>
```
Comment (Nkemjiks) [inline src/components/SignupForm/SignupForm.js]: Spell check in title

File: src/components/SignupForm/SignupForm.js [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+import Input from '../Input/Input';
+import validation from './validation';
+import AuthModal from '../AuthModal/AuthModal';
+import SocialLogin from '../SocialLogin/SocialLogin';
+import Separator from '../Separator/Separator';
+import Aux from '../Aux/Aux';
+import './SignupForm.scss';
+
+const SignupForm = () => {
+  const [formdata, setFormData] = useState([
```
Comment (Nkemjiks) [inline src/components/SignupForm/SignupForm.js]: Move from here to line 88 to a new file

File: src/components/SignupForm/SignupForm.js [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+            <button type="submit" className="btn">{resourceLoading ? <i className="fa fa-spinner fa-spin loader" /> : 'signup'}</button>
+            <p className="small-text">
+          Already have an account?
+              {' '}
+              <a href="abc">Sign In</a>
+            </p>
+          </form>
+        </AuthModal>
+      ) : (
+        <AuthModal desc="we have sent link to your email to verify your account, please click the link to verify your account " title="Account created" isOpen>
```
Comment (Nkemjiks) [inline src/components/SignupForm/SignupForm.js]: Capitalise where necessary in the description

**Review by Nkemjiks — CHANGES_REQUESTED** [review]

File: src/components/App.js (hunk 1 above)
Comment (WilliamsOhworuka) [reply src/components/App.js]: done

File: src/components/App.js (hunk 2 above)
Comment (WilliamsOhworuka) [reply src/components/App.js]: done

File: src/components/AuthModal/AuthModal.js (hunk 3 above)
Comment (WilliamsOhworuka) [reply src/components/AuthModal/AuthModal.js]: done

File: src/components/AuthModal/AuthModal.js (hunk 4 above)
Comment (WilliamsOhworuka) [reply src/components/AuthModal/AuthModal.js]: done

File: src/components/AuthModal/AuthModal.scss (hunk 5 above)
Comment (WilliamsOhworuka) [reply src/components/AuthModal/AuthModal.scss]: done

File: src/components/SignupForm/SignupForm.js (hunk 6 above)
Comment (WilliamsOhworuka) [reply src/components/SignupForm/SignupForm.js]: done

File: src/components/SignupForm/SignupForm.js (hunk 7 above)
Comment (WilliamsOhworuka) [reply src/components/SignupForm/SignupForm.js]: done

File: src/components/SignupForm/SignupForm.js (hunk 8 above)
Comment (WilliamsOhworuka) [reply src/components/SignupForm/SignupForm.js]: done

**Pull request merged**